File: solutions/2026-07-14/solution.py
# Maximum Subarray Sum
# Difficulty: Medium
# URL: https://leetcode.com/problems/maximum-subarray-sum/

def maximumSum(nums):
    if len(nums) < 3:
        return -float('inf')
    max_sum = float('-inf')
    for i in range(len(nums) - 2):
        max_sum = max(max_sum, nums[i] + nums[i+1] + nums[i+2])
    return max_sum

File: solutions/2026-07-14/solution.java
// Maximum Subarray Sum
// Difficulty: Medium
// URL: https://leetcode.com/problems/maximum-subarray-sum/

class Solution {
    public int maximumSum(int[] nums) {
        if (nums.length < 3) return -Integer.MAX_VALUE;
        int max_sum = Integer.MIN_VALUE;
        for (int i = 0; i < nums.length - 2; i++) {
            max_sum = Math.max(max_sum, nums[i] + nums[i+1] + nums[i+2]);
        }
        return max_sum;
    }
}